src/main/ruby/truffleruby/core/truffle/feature_loader.rb

```diff
@@ -292,13 +305,24 @@ def self.get_loaded_features_index
       unless @loaded_features_version == $LOADED_FEATURES.version
         raise '$LOADED_FEATURES is frozen; cannot append feature' if $LOADED_FEATURES.frozen?
         @loaded_features_index.clear
+        previous_realpaths = @features_realpath_cache.dup
+        @features_realpath_cache.clear
+        @loaded_features_realpaths.clear
```

Comment:
Maybe this cache never needs to be cleared.

Reply:
I think it was necessary to satisfy the specs, and I saw the CRuby code doing something simliar:
https://github.com/ruby/ruby/blob/7ce4b716bdb5bcfc8b30ffcd034ce7aded1f72b9/load.c#L363-L364
https://github.com/ruby/ruby/blob/7ce4b716bdb5bcfc8b30ffcd034ce7aded1f72b9/load.c#L377-L384